tenant_users/tenants/tasks.py

```diff
@@ -43,11 +51,20 @@ def provision_tenant(tenant_name, tenant_slug, user_email):
         )
 
         # Add user as a superuser inside the tenant
-        tenant.add_user(user, is_superuser=True)
-    except:
-        if tenant is not None:
-            #Flag is set to auto-drop the schema for the tenant
-            tenant.delete(True)
-        raise
+    else:  # django-tenants
+        tenant = TenantModel.objects.create(name=tenant_name,
+                                            slug=tenant_slug,
+                                            schema_name=schema_name,
+                                            owner=user)
+
+        # Add one or more domains for the tenant
+        domain = get_tenant_domain_model().objects.create(domain=tenant_domain,
+                                                          tenant=tenant,
+                                                          is_primary=True)
+
+    tenant.add_user(user, is_superuser=True, is_staff=is_staff)
+    # if tenant is not None:
+    #     #Flag is set to auto-drop the schema for the tenant
+    #     tenant.delete(True)
```

Comment:
```
    except:
        if tenant is not None:
            #Flag is set to auto-drop the schema for the tenant
            tenant.delete(True)
        raise
```

This catch-all exception handler actually re-raises the exception -- thus no exceptions are getting masked (lost) by the catch all when one happens.

We have seen migrations that worked but due to some other state in the system would fail during tenant creation. One other case where this can potentially happen is if a child of tenant overrides add_user and causes issues there. The user may not end up linked to the tenant but the tenant will still be provisioned. This state can be recovered from later, manually, but IMO it's best to fail hard early on, wipe the tenant, and let the exception bubble up.

If any of those cases occur, we now get the best of both worlds. First, the exception bubbles up and if it doesn't get handled, it gets logged (i.e. to Sentry). Second, we also keep the DB clean from tenants that didn't fully provision.